internal/plugin/vault_config_reporting.go

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package plugin
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+type vaultReportingLicenseConfig struct {
```

Comment:
Did we try to use a [dynamicPseudoTypeBlock](https://github.com/hashicorp-forge/terraform-provider-enos/blob/main/internal/plugin/tftypes.go#L1185) for this instead of creating an individual type? I suppose the benefit here is having named attributes in a struct vs keys in a map, so if that's something we need then I understand, but at first glance it feels largely the same.

If we do end up keeping this I want to make sure that all of none of the keys are required in tests. So we'll want to test setting no reporting config, all of it, and some of it.